File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.java
package com.example.todonotesapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.todonotesapp.MyNotesActivity;
import com.example.todonotesapp.R;
import com.example.todonotesapp.clickListeners.ItemClickListener;
import com.example.todonotesapp.model.Notes;

import java.util.List;

public class NotesAdapter  extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {
    private List<Notes> listNotes;
   private ItemClickListener itemClickListener;
    public NotesAdapter(List<Notes> list,ItemClickListener itemClickListener){
    this.listNotes = list;
    this.itemClickListener=itemClickListener;
    }
    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout, parent,false);
        return new ViewHolder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
    //bind data from list to ViewHolder
        final Notes notes =listNotes.get(position);
        String title = notes.getTitle();
        String description = notes.getDescription();

        holder.TextViewTitle.setText(title);
        holder.TextViewDescription.setText(description);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                itemClickListener.onClick(notes);
            }
        });
    }

    @Override
    public int getItemCount() {
        return listNotes.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView TextViewTitle,TextViewDescription;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            TextViewTitle = itemView.findViewById(R.id.textViewTitle);
            TextViewDescription = itemView.findViewById(R.id.textViewDescription);
        }
    }
}

File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.kt
package com.example.todonotesapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.view.menu.MenuView
import androidx.recyclerview.widget.RecyclerView
import com.example.todonotesapp.R
import com.example.todonotesapp.clickListeners.ItemClickListener
import com.example.todonotesapp.model.Notes

class NotesAdapter(val  list:List<Notes>, val itemClickListener: ItemClickListener) :RecyclerView.Adapter<NotesAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout,parent,false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return  list.size
    }

    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {
        val notes = list[position]
        val title =  notes.title
        val description = notes.description
        holder.textViewTitle.text = title
        holder.textViewDescription.text = description
        holder.itemView.setOnClickListener(object : View.OnClickListener{
            override fun onClick(v: View?) {
                itemClickListener.onClick(notes)
            }

        })
    }
    inner class ViewHolder(itemView:View):RecyclerView.ViewHolder(itemView){
        val textViewTitle :TextView = itemView.findViewById(R.id.textViewTitle)
        val textViewDescription:TextView = itemView.findViewById(R.id.textViewDescription)
    }
}